lnst/Recipes/ENRT/SRIOVNetnsBridgeRecipe.py

```diff
@@ -0,0 +1,195 @@
+import time
+
+from lnst.Common.Parameters import Param
+from lnst.Common.IpAddress import ipaddress
+from lnst.Controller import HostReq, DeviceReq, RecipeParam
+from lnst.Controller.NetNamespace import NetNamespace
+from lnst.Recipes.ENRT.BaremetalEnrtRecipe import BaremetalEnrtRecipe
+from lnst.RecipeCommon.Ping.PingEndpoints import PingEndpoints
+from lnst.Recipes.ENRT.ConfigMixins.OffloadSubConfigMixin import (
+    OffloadSubConfigMixin,
+)
+from lnst.Recipes.ENRT.ConfigMixins.CommonHWSubConfigMixin import (
+    CommonHWSubConfigMixin,
+)
+from lnst.Devices import BridgeDevice
+
+
+class SRIOVNetnsBridgeRecipe(
+    CommonHWSubConfigMixin, OffloadSubConfigMixin, BaremetalEnrtRecipe
+):
+    """
+    This recipe implements Enrt testing for a SRIOV network scenario
+    with VF located in the network namespace to mimic container network.
+    PF with VF representor is plugged into the Linux bridge
+    and setups looks following.
+
+    .. code-block:: none
+
+                      +--------+
+               +------+ switch +-------+
+               |      +--------+       |
+       +-------|------+        +-------|------+
+       |    +--|--+   |        |    +--|--+   |
+    +--|----|eth0|--- |--+  +--|----|eth0|--- |--+
+    |  |    +----+    |  |  |  |    +----+    |  |
+    |  |       |      |  |  |  |       |      |  |
+    |  |vf_representor|  |  |  |vf_representor|  |
+    |  |              |  |  |  |              |  |
+    |  +-Linux bridge-+  |  |  +-Linux bridge-+  |
+    |         |          |  |         |          |
+    |    +-namespace-+   |  |    +-namespace-+   |
+    |   |    vf0     |   |  |   |    vf0     |   |
+    |   +-----------+    |  |   +-----------+    |
+    |      host1         |  |       host2        |
+    +--------------------+  +--------------------+
+
+    All sub configurations are included via Mixin classes.
+
+    The actual test machinery is implemented in the :any:`BaseEnrtRecipe` class.
+    """
+    host1 = HostReq()
+    host1.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    host2 = HostReq()
+    host2.eth0 = DeviceReq(label="net1", driver=RecipeParam("driver"))
+
+    vf_suffix = StrParam(default="_0")
```

Comment:
as discussed on irc it should be explained that this requires the use of specific kernel boot parameters to work as expected.